File: projects/macrotool/src/app/shared/models/Employee.ts
export class Employee {
    id?: string;
    cuil: number = 0;
    name: string = '';
    employeeID: number = 0;
    isOperational: boolean = true;
    sector!: ESector;

    documentType: string = '';
    documentNumber: string = '';
    birthDate: string = '';
    gender: EGender = EGender.MALE;
    civilStatus: ECivilStatus = ECivilStatus.SINGLE;
    nationality: string = '';

    adress?: string;
    city: string = '';
    province: string = '';
    country: string = '';
    zipCode: string = '';

    phone?: string;
    cellPhone?: string;
    email: string = '';

    entryDate?: string;
    leaveDate?: string;

    constructor(init?: Partial<Employee>) {
        Object.assign(this, init);
    }
}

export enum EGender {
    MALE = 'M',
    FEMALE = 'F',
    OTHER = 'O'
}

export enum ECivilStatus {
    SINGLE = 'Soltero',
    MARRIED = 'Casado',
    DIVORCED = 'Divorciado',
    WIDOWED = 'Viudo',
    OTHER = 'Otro'
}


export enum ESector {
    DESMALEZADO,
    CLEANING_OPERATOR,
    ADMINISTRATION
}

File: src/main/java/com/fluxenture/core/employees/domain/Employee.java
package com.fluxenture.core.employees.domain;

import com.fluxenture.core.shared.domain.AuditMetadata;
import jakarta.annotation.Nullable;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Employee {
    @Nullable
    private String id;
    private long cuil;
    private String name;
    private int employeeID; // Legajo
    private Boolean isOperational = true;
    private ESector sector;
    private EService service;

    // --- Identidad y Estado Civil ---
    private String documentType;
    private String documentNumber;
    private LocalDate birthDate;    // Fecha de Nacimiento
    private String gender;       // Sexo
    private String civilStatus;  // Estado Civil
    private String nationality;  // Nacionalidad

    // --- Ubicación y Contacto ---
    @Nullable
    private String adress;       // Domicilio
    private String city;         // Localidad
    private String province;     // Provincia
    private String country;      // País
    private String zipCode;      // Código Postal

    @Nullable
    private String phone;        // Teléfono fijo
    @Nullable
    private String cellPhone;    // Celular
    private String email;        // Correo Electrónico

    // --- Fechas de Contrato ---
    @Nullable
    private LocalDate entryDate;    // Fecha de alta
    @Nullable
    private LocalDate leaveDate;    // Fecha de baja

    // --- Auditoría ---
    private AuditMetadata audit;
}
